backend/typescript/rest/campRoutes.ts

```diff
@@ -19,25 +29,35 @@ campRouter.get("/", async (req, res) => {
 });
 
 /* Create a camp */
-campRouter.post("/", createCampDtoValidator, async (req, res) => {
-  try {
-    const newCamp = await campService.createCamp({
-      ageLower: req.body.ageLower,
-      ageUpper: req.body.ageUpper,
-      name: req.body.name,
-      description: req.body.description,
-      location: req.body.location,
-      capacity: req.body.capacity,
-      fee: req.body.fee,
-      formQuestions: req.body.formQuestions,
-      campSessions: req.body.campSessions,
-    });
-
-    res.status(201).json(newCamp);
-  } catch (error: unknown) {
-    res.status(500).json({ error: getErrorMessage(error) });
-  }
-});
+campRouter.post(
+  "/",
+  upload.single("file"),
+  createCampDtoValidator,
+  async (req, res) => {
+    try {
+      const body = JSON.parse(req.body.body);
+      const newCamp = await campService.createCamp({
+        ageLower: body.ageLower,
+        ageUpper: body.ageUpper,
+        name: body.name,
+        description: body.description,
+        location: body.location,
+        capacity: body.capacity,
+        fee: body.fee,
+        formQuestions: body.formQuestions,
+        campSessions: body.campSessions,
+        filePath: req.file?.path,
+        fileContentType: req.file?.mimetype,
+      });
+      if (req.file?.path) {
+        fs.unlinkSync(req.file.path);
```

Comment:
what is this doing?

Reply:
Each image thats uploaded is temporarily stored in "upload" directory, so this `fs.unlinkSync(req.file.path)` deletes the file from that directory. For example, req.file.path is something like 'uploads/152280faf4e59f64175f8a7186a48eab', so `fs.unlinkSync(req.file.path)` will just remove `152280faf4e59f64175f8a7186a48eab` from the upload directory

Reply:
ah ok gotcha